e2e/lockedFields.e2e.spec.ts

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright © 2023 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * Integration tests that verify that the deployment profile reader is finding locked fields,
+ * and that rdctl can't change those locked preferences.
+ */
+
+import os from 'os';
+import path from 'path';
+
+import { expect, test } from '@playwright/test';
+
+import { NavPage } from './pages/nav-page';
+import { createDefaultSettings, createUserProfile, startRancherDesktop, teardown } from './utils/TestUtils';
+
+import type { DeploymentProfileType } from '@pkg/config/settings';
+import { readDeploymentProfiles } from '@pkg/main/deploymentProfiles';
+import { spawnFile } from '@pkg/utils/childProcess';
+
+import type { ElectronApplication, BrowserContext, Page } from '@playwright/test';
+
+test.describe('Command server', () => {
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+  let page: Page;
+  const appPath = path.join(__dirname, '../');
+  let deploymentProfile: DeploymentProfileType|null = null;
+
+  function rdctlPath() {
+    return path.join(appPath, 'resources', os.platform(), 'bin', os.platform() === 'win32' ? 'rdctl.exe' : 'rdctl');
+  }
+
+  async function rdctl(commandArgs: string[]): Promise< { stdout: string, stderr: string, error?: any }> {
+    try {
+      return await spawnFile(rdctlPath(), commandArgs, { stdio: 'pipe' });
+    } catch (err: any) {
+      return {
+        stdout: err?.stdout ?? '', stderr: err?.stderr ?? '', error: err,
+      };
+    }
+  }
+
+  function saveUserProfile() {
+    // If there's an error in this code, let it bubble up for the caller to deal with it.
+    deploymentProfile = readDeploymentProfiles();
+  }
+
+  async function restoreUserProfile() {
+    // the test for a non-null is to satisfy typescript
```

Comment:
That's because we could have a default (in case `saveUserProfile` throws):
```ts
  let deploymentProfile: DeploymentProfileType = { defaults: {}, locked: {} };
```

Reply:
Ah right.

Reply:
FWIW, this means if we don't allow `deploymentProfile` to be `null`, we can drop the check.